lang/cs/Org.Apache.REEF.Tests/Functional/Bridge/TestCloseTask.cs

```diff
@@ -322,6 +386,85 @@ public void OnError(Exception error)
             }
         }
 
+        private sealed class BreakTestTask : ITask, IObserver<ICloseEvent>
+        {
+            private volatile bool _break;
+            private volatile bool _stoped;
+            private readonly bool _enforceClose;
+            private readonly int _enforceCloseTimeout;
+
+            [Inject]
+            private BreakTestTask(
+                [Parameter(typeof(EnforceCloseTimeout))] int enforceCloseTimeout,
+                [Parameter(typeof(EnforceClose))] bool enforceClose)
+            {
+                _enforceClose = enforceClose;
+                _enforceCloseTimeout = enforceCloseTimeout;
+            }
+
+            public byte[] Call(byte[] memento)
+            {
+                int iterate = 1;
+
+                while (!_break && iterate < 100)
```

Comment:
Our tests should be deterministic and should never rely upon iteration count and timeouts.